allennlp/models/semantic_role_labeler.py

```diff
@@ -175,6 +180,16 @@ def tag(self, text_field: TextField, verb_indicator: IndexField) -> Dict[str, An
 
         return {"tags": tags, "class_probabilities": predictions.numpy()}
 
+    def get_metrics(self, reset: bool = False):
+        metric_dict = self.span_metric.get_metric(reset=reset)
+        if self.training:
+            # This can be a lot of metrics, as there are 3 per class.
+            # During training, we only really care about the overall
+            # metrics, so we filter for them here.
+            return {x: y for x, y in metric_dict if "overall" in x}
```

Comment:
I don't think this is actually going to work - you'll get different metrics for validation and train, and so the training code will crash.

Hmm...  maybe not, because the training code uses the keys in the training metric dict to print out both validation and training metrics...  Ok.  I don't think this is ideal, but the other solutions are ugly too, and this is probably good enough for now.  This could very easily break in the future, though.